app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt

```diff
@@ -212,6 +214,7 @@ class DefaultSessionControlController(
             tab,
             onTabRestored = {
                 activity.openToBrowser(BrowserDirection.FromHome)
+                reloadUrlUseCase.invoke(sessionManager.selectedSession)
```

Comment:
Can you help me understand why this is needed? It somewhat defeats the purpose of the tab getting restored in its previous state (which is weird at times, but if we do not want that then we should change that feature).

Reply:
Yes! According to @liuche's [comment](https://github.com/mozilla-mobile/fenix/issues/10417#issuecomment-714666960) on the issue, we want to keep the backstack of the page at the time that it was added to the collection, but have a fresh version of the page (e.g. if it is a news site, we should see the latest news not the news at the time we added to the collection). To keep the backstack, we `restore` the page; then we `reload` to make it up-to-date.

Does that make sense?